api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +566,135 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+    public static final String DEFAULT_CRS = "OGC:CRS84";
+
+    public static GeometryType crs84() {
+      return new GeometryType();
+    }
+
+    public static GeometryType of(String crs) {
+      return new GeometryType(crs);
+    }
+
+    private final String crs;
+
+    private GeometryType() {
+      crs = null;
+    }
+
+    private GeometryType(String crs) {
+      Preconditions.checkArgument(crs == null || !crs.isEmpty(), "Invalid CRS: (empty string)");
+      this.crs = DEFAULT_CRS.equalsIgnoreCase(crs) ? null : crs;
+    }
+
+    @Override
+    public TypeID typeId() {
+      return TypeID.GEOMETRY;
+    }
+
+    public String crs() {
+      return crs;
```

Comment:
If CRS is null, should this return `DEFAULT_CRS`?

Reply:
I think it is better to return `DEFAULT_CRS`, so that the callers don't have to handle both nulls and default CRS.

We can make `GeographyType.algorithm` return `EdgeAlgorithm.SPHERICAL` when algorithm is null as well.